File: Day3/reverse_number.py
""" Program to reverse the Number """

def get_reverse_number(num):

    """ To handle the Negative number """

    if num < 0:
        num = -num
        reversed_num = get_reverse_number(num)
        reversed_num = -reversed_num
        return reversed_num
    reversed_num = 0

    while num !=0:
        reversed_num = (reversed_num *10) + (num % 10)
        num = num //10
    return reversed_num


original_num = int(input("Enter the number :"))

reversedNum = get_reverse_number(original_num)

print("The Reverse of a number is :",reversedNum)

File: Day3/ReverseNumber.cpp
#include<iostream>
using namespace std;

int getReverseNumber(int num){
    // To handle the negative number
    if(num <0){
        num = -num;
        int reversedNum = getReverseNumber(num);
        reversedNum = -reversedNum;
        return reversedNum;
    }
    // For positive integer number
    int reversedNum = 0;
    while(num != 0){
        reversedNum = (reversedNum* 10) + (num% 10);
        num = num / 10;
    }
    return reversedNum;
}
int main(){
    
    int originalNum;
    cout<<"Enter the Number :";
    cin>>originalNum;

    int reversedNum  =  getReverseNumber(originalNum);
    cout<<"The reverse number of a given number is : "<<reversedNum;
    return 0;
}

